src/headers.js

```diff
@@ -10,13 +10,17 @@ class Headers {
   constructor (init) {
     this[kHeaders] = new Map()
 
-    if (init) {
+    if (init && typeof init === 'object') {
       if (Array.isArray(init)) {
         for (let i = 0, header = init[0]; i < init.length; i++, header = init[i]) {
           if (header.length !== 2) throw TypeError('header entry must be of length two')
           this.append(header[0], header[1])
         }
-      } else if (typeof init === 'object' && !types.isBoxedPrimitive(init)) {
+      } else if (kHeaders in init) {
+        for (const [name, values] of init[kHeaders]) {
+          this[kHeaders].set(name, values.slice())
```

Comment:
I wasn't sure here if I should follow the pattern of using prototype methods. Is there any specific reason that is done here?